torchgeo/datasets/chabud.py

```diff
@@ -0,0 +1,294 @@
+# Licensed under the MIT License.
+
+"""ChaBuD dataset."""
+
+import os
+from typing import Callable, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from .geo import NonGeoDataset
+from .utils import DatasetNotFoundError, download_url, percentile_normalization
+
+
+class ChaBuD(NonGeoDataset):
+    """ChaBuD dataset.
+
+    The `ChaBuD <https://huggingface.co/spaces/competitions/ChaBuD-ECML-PKDD2023>`__
+    dataset is a dataset for Change detection for Burned area Delineation and is used
+    for the ChaBuD ECML-PKDD 2023 Discovery Challenge.
+
+    Dataset features:
+
+    * Sentinel-2 multispectral imagery
+    * binary masks of burned areas
+    * 12 multispectral bands
+    * 356 pairs of pre and post images with 10 m per pixel resolution (512x512 px)
+
+    Dataset format:
+
+    * single hdf5 dataset containing images and masks
+
+    Dataset classes:
+
+    0. no change
+    1. burned area
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://doi.org/10.1016/j.rse.2021.112603
+
+    .. note::
+
+       This dataset requires the following additional library to be installed:
+
+       * `h5py <https://pypi.org/project/h5py/>`_ to load the dataset
+
+    .. versionadded:: 0.6
+    """
+
+    all_bands = [
+        "B1",
+        "B2",
+        "B3",
+        "B4",
+        "B5",
+        "B6",
+        "B7",
+        "B8",
+        "B8A",
+        "B9",
+        "B11",
+        "B12",
+    ]
+    rgb_bands = ["B4", "B3", "B2"]
+    folds = {"train": [1, 2, 3, 4], "val": [0]}
+    url = "https://huggingface.co/datasets/chabud-team/chabud-ecml-pkdd2023/resolve/main/train_eval.hdf5"  # noqa: E501
+    filename = "train_eval.hdf5"
+    md5 = "15d78fb825f9a81dad600db828d22c08"
+
+    def __init__(
+        self,
+        root: str = "data",
+        split: str = "train",
+        bands: list[str] = rgb_bands,
+        transforms: Optional[Callable[[dict[str, Tensor]], dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new ChaBuD dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train" or "val"
+            bands: the subset of bands to load
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: if ``split`` or ``bands`` argument are invalid
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+        """
+        assert split in self.folds
+        for band in bands:
+            assert band in self.all_bands
```

Comment:
Could be done in one line if you want:
```suggestion
        assert set(bands) <= set(self.all_bands)
```
This is optional though, whatever you think is clearer.